File: app/src/main/java/home/self/beerviewer_mvvm/view/beersview/BeersViewModule.java
package home.self.beerviewer_mvvm.view.beersview;

import dagger.Module;
import dagger.Provides;
import home.self.beerviewer_mvvm.data.source.BeerRepository;
import home.self.beerviewer_mvvm.di.ActivityScope;
import home.self.beerviewer_mvvm.rx.schedulers.BaseSchedulerProvider;

/**
 * Created by Tak on 2018. 4. 19..
 */

@Module
public class BeersViewModule {


    @Provides
    @ActivityScope
    BeersViewModel provideBeersViewModel(BeerRepository beerRepository, BaseSchedulerProvider schedulerProvider) {
        return new BeersViewModel(beerRepository, schedulerProvider);
    }
}

File: app-kotlin/src/main/java/home/self/beerviewer_mvvm/app_kotlin/view/beersview/BeersViewModule.kt
package home.self.beerviewer_mvvm.app_kotlin.view.beersview

import androidx.lifecycle.ViewModel
import com.mashup.dutchmarket.di.key.ViewModelKey
import dagger.Binds
import dagger.Module
import dagger.Provides
import dagger.multibindings.IntoMap

/**
 * Created by Tak on 2018. 5. 16..
 */

@Module(includes = arrayOf(BeersViewModule.ProvideModule::class))
internal interface BeersViewModule {

    @Module
    class ProvideModule {

        @Provides
        fun provideBeersAdapter(activity: BeersViewActivity) : BeersAdapter = BeersAdapter().apply { setOnItemClickListener(activity) }
    }

    @Binds
    @IntoMap
    @ViewModelKey(BeersViewModel.ViewModel::class)
    fun bindBeersViewModel(beersViewModel: BeersViewModel.ViewModel): ViewModel
}